### ISSUE TITLE
GraphRequestAdapter must be a subclass of SerializationWriterFactory::class

### ISSUE THREAD

**Issue by uncaught** [microsoftgraph/msgraph-sdk-php#1523]
Your version 2.8.0 introduced a breaking bug!

```
InvalidArgumentException: The class passed must be a subclass of SerializationWriterFactory::class
/var/www/html/vendor/microsoft/kiota-abstractions/src/ApiClientBuilder.php:25
/var/www/html/vendor/microsoft/microsoft-graph/src/Generated/BaseGraphClient.php:592
/var/www/html/vendor/microsoft/microsoft-graph/src/GraphServiceClient.php:45
/var/www/html/vendor/microsoft/microsoft-graph/src/GraphServiceClient.php:70
/var/www/html/vendor/microsoft/microsoft-graph/src/GraphServiceClient.php:85
```

Apparently your own `GraphRequestAdapter` does not extend `SerializationWriterFactory` as requested!

Please write tests.

**Comment by dominicgroza72** [comment]
We need this fixed too...

**Comment by SilasKenneth** [comment]
Hi @uncaught, thank you for reporting the issue and I'm sorry for the inconvenience caused.
Are you still experiencing the issue?
If yes, kindly try doing a composer update to update the dependencies required by the SDK. The issue was supposed to be addressed by [This PR](https://github.com/microsoftgraph/msgraph-sdk-php-core/pull/173).
If the issue is gone, please consider closing the issue.

**Comment by uncaught** [comment]
Hey @SilasKenneth, that PR seems to fix the issue.

However, composer won't update by itself to `"microsoft/microsoft-graph-core": "^2.1.1"` because that would require a downgrade of `php-http/promise` from 1.3 to 1.2. Even with `composer update -W` you won't get 2.1.1. I had to manually add  `"microsoft/microsoft-graph-core": "^2.1.1"` to my composer.json, which is not a good sign:

![image](https://github.com/microsoftgraph/msgraph-sdk-php/assets/10322643/f335e658-af8c-49b7-b9bc-227cb1431321)

You need to release a new bugfix version of this library, setting `"microsoft/microsoft-graph-core": "^2.1.1"` as dependency to solve this issue at least.

Then another issue should be added with `microsoft/kiota-abstractions` as to why they need `"php-http/promise": "~1.2.0"` and cannot go to 1.3.

**Comment by SilasKenneth** [comment]
Hi @uncaught,
The version bump is included in the next version 2.9.0. 
For the `php-http/promise`, we fixed it to that version since we currently depend on the ability to add a Generic type to promise in the doc comment which helps with resolving the type that is return by `.wait()` on a promise.

**Comment by teemukolehmainen-howspace** [comment]
This same issue is happening with `"microsoft/microsoft-graph": "^2.10.0"` now.

**Comment by aless673** [comment]
why this issue is closed ? I just installed this package and did the quickstart, and it immediatly fails with this error

**Comment by nusagates** [comment]
i just installed this package today and facing the same issue.

**Comment by MaximilianGewers** [comment]
Running into the same issue after updating to the latest `#/components/schemas/microsoft.graph.user` schema via kiota generate.

If anybody is following [Build API clients for PHP with Microsoft identity authentication](https://learn.microsoft.com/en-us/openapi/kiota/tutorials/php-azure?tabs=portal) and running into the same issues (thats why I am here) it seems that the instructions that talk about [adding dependencies](https://learn.microsoft.com/en-us/openapi/kiota/tutorials/php-azure?tabs=portal#add-dependencies) are outdated. They are missing:

```
composer require microsoft/kiota-serialization-multipart
composer require microsoft/kiota-serialization-form
```

After adding these two my previously working login is working again.

**Comment by Casmo** [comment]
For this still running into this issue. Set your version fixed on v2.7.0 in `composer.json`.

```json
{
  "require": {
    "microsoft/microsoft-graph": "2.7.0",
  }
}
```

**Comment by flyke** [comment]
Thank you [MaximilianGewers](https://github.com/MaximilianGewers). I got this error today when following [this Microsoft Tutorial](https://learn.microsoft.com/en-us/graph/tutorials/php-app-only?tabs=aad&tutorial-step=3) step by step.

Your composer required lines fixed the error for me.
Super weird to follow a Microsoft tutorial to the letter and immediatly running into a bug.
